.github/workflows/on-pre-release.yaml

```diff
@@ -0,0 +1,28 @@
+name: Pre-Release CI
+
+on:
+  release:
+    types:
+      - prereleased
+
+jobs:
+  run-unit-tests:
+    name: Run Unit Tests
+    uses: ./.github/workflows/test.yaml
+
+  publish-docker-image:
+    name: Publish Pre-Release Docker Image
+    uses: ./.github/workflows/publish-docker-images.yaml
+    secrets: inherit
+    needs:
+      - run-unit-tests
+    permissions:
+      contents: read
+      packages: write
```

Comment:
> FYI For pre-releases, it's important to append the -rc and set it as a pre-release:

Do you think its possible to add that as a rule ? Like if there's no "-rc" then fail ?

Reply:
> > FYI For pre-releases, it's important to append the -rc and set it as a pre-release:
> 
> Do you think its possible to add that as a rule ? Like if there's no "-rc" then fail ?

Yepp, I added it in latest commit

Reply:
💪